packages/flame_isolate/example/lib/standard/int_vector2.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:flame/components.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:quiver/core.dart';
+
+@immutable
+class IntVector2 {
+  final int x;
+  final int y;
+
+  const IntVector2(this.x, this.y);
+
+  /// Manhattan distance on a square grid
+  int distanceTo(IntVector2 b) {
+    return (x - b.x).abs() + (y - b.y).abs();
+  }
+
+  @override
+  bool operator ==(Object o) => o is IntVector2 && x == o.x && y == o.y;
```

Comment:
Since this class is const, I think this is already automatically done.

Reply:
Seems like it is not, actually. Get a `Null check operator used on a null value` when trying to run the game if it is not present.